kontrol/keypair.go

```diff
@@ -0,0 +1,110 @@
+package kontrol
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/koding/cache"
+)
+
+// KeyPair defines a single key pair entity
+type KeyPair struct {
+	// ID is the unique id defining the key pair
+	ID string
+
+	// Public key is used to validate tokens
+	Public string
+
+	// Private key is used to sign/generate tokens
+	Private string
+}
+
+func (k *KeyPair) Validate() error {
+	if k.ID == "" {
+		return errors.New("KeyPair ID field is empty")
+	}
+
+	if k.Public == "" {
+		return errors.New("KeyPair Public field is empty")
+	}
+
+	if k.Private == "" {
+		return errors.New("KeyPair Private field is empty")
+	}
+	return nil
+}
+
+// KeyPairStorage is responsible of managing key pairs
+type KeyPairStorage interface {
+	// AddKey adds the given key pair to the storage
```

Comment:
It was like that previously. But Postgres already has `Delete` or `Add` methods. That's why I changed them.